aggoracle/chaingersender/evm.go

```diff
@@ -41,48 +64,92 @@ type EVMChainGERSender struct {
 func NewEVMChainGERSender(
 	logger *log.Logger,
 	l2GERManagerAddr common.Address,
+	aggOracleCommitteeAddr common.Address,
 	l2Client aggkittypes.BaseEthereumClienter,
 	ethTxMan types.EthTxManager,
 	gasOffset uint64,
 	waitPeriodMonitorTx time.Duration,
+	enableAggOracleCommittee bool,
 ) (*EVMChainGERSender, error) {
+	// Determine mode based on configuration
+	mode := DirectInjectionMode
+	if enableAggOracleCommittee {
+		mode = AggOracleCommitteeMode
+	}
+
+	// Always initialize L2 GER Manager (needed for checking if GER is injected)
 	l2GERManager, err := globalexitrootmanagerl2sovereignchain.NewGlobalexitrootmanagerl2sovereignchain(
 		l2GERManagerAddr, l2Client)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create binding for GER L2 manager (SC address: %s): %w", l2GERManagerAddr, err)
 	}
 
-	if err := validateGERSender(ethTxMan.From(), l2GERManager); err != nil {
-		return nil, err
-	}
-
 	l2GERAbi, err := globalexitrootmanagerl2sovereignchain.Globalexitrootmanagerl2sovereignchainMetaData.GetAbi()
 	if err != nil {
 		return nil, fmt.Errorf("failed to retrieve GER L2 manager ABI: %w", err)
 	}
 
-	return &EVMChainGERSender{
-		logger:              logger,
-		l2GERManager:        l2GERManager,
-		l2GERManagerAddr:    l2GERManagerAddr,
-		l2GERManagerAbi:     l2GERAbi,
-		ethTxMan:            ethTxMan,
-		gasOffset:           gasOffset,
-		waitPeriodMonitorTx: waitPeriodMonitorTx,
-	}, nil
+	sender := &EVMChainGERSender{
+		logger:                 logger,
+		mode:                   mode,
+		l2GERManager:           l2GERManager,
+		l2GERManagerAddr:       l2GERManagerAddr,
+		l2GERManagerAbi:        l2GERAbi,
+		aggOracleCommitteeAddr: aggOracleCommitteeAddr,
+		l2Client:               l2Client,
+		ethTxMan:               ethTxMan,
+		gasOffset:              gasOffset,
+		waitPeriodMonitorTx:    waitPeriodMonitorTx,
+	}
+
+	// Initialize mode-specific components
+	if err := sender.initializeMode(); err != nil {
+		return nil, err
+	}
+
+	logger.Infof("EVMChainGERSender initialized in %s mode", mode)
+	return sender, nil
 }
 
-// validateGERSender validates whether the provided GER sender is allowed to send and remove GERs
-func validateGERSender(gerSender common.Address, l2GERManagerSC types.L2GERManagerContract) error {
-	zeroAddr := common.Address{}
-	gerUpdater, err := l2GERManagerSC.GlobalExitRootUpdater(nil)
+// initializeMode initializes mode-specific components and validations
+func (c *EVMChainGERSender) initializeMode() error {
+	switch c.mode {
+	case DirectInjectionMode:
+		return c.initializeDirectInjectionMode()
```

Comment:
We probably don't need the `initializeDirectInjectionMode`, and can be simplified this way. WDYT?

```suggestion
		return validateGERSender(c.ethTxMan.From(), c.l2GERManager)
```

Reply:
done